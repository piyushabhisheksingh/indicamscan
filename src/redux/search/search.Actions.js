import * as types from './search.Types';
import { stocksprefix } from '../../utils/commonFunctions'

/**
 * searchAction, it takes 3 params and calls the search api function
 */
export const search = (index, input, prevIndex, delay) => {
    /**
    * @param {number|string|number} index input prevIndex - index is the current selectedTab, input is the user search query and prevIndex of selected tab
    */
    let params = stocksprefix(index, input);
    return {
        type: types.SEARCH,
        params,
        scrip: input,
        mode: index,
        prevIndex,
        delay
    }
}

/**
 * clearSearchState, this will not take any argument and it clears the search text input
 */
export const clearSearchState = () => {
    return {
        type: types.CLEAR_SEARCH_STATE
    }
}

/**
 * recentSearch, this will not take any argument and it will set a state for recent search
 */
// export const recentSearch = () => {
//     return {
//         type: types.RECENT_SEARCH
//     }
// }

/**
 * setSearchText, this will take one argument and sets the user search input
 */
export const setSearchText = (payload, key) => {
    /**
    * @param {string} payload - payload is the user search input
    */
    return {
        type: types.SET_SEARCH_TEXT,
        payload,
        key
    }
}

/**
 * setTab, this will take one argument and sets the selected tab
 */
export const setTab = (payload) => {
    /**
   * @param {number} payload - payload is the user selectedTabIndex
   */
    return {
        type: types.SET_SELECTED_TAB,
        payload
    }
}

/**
 * clearState, this will not take any arguments and it clear the result array
 */
export const clearState = () => {
    return {
        type: types.CLEAR_RESULT
    }
}

export const addStock = (payload, onSuccess, onError) => {
    return {
        type: types.ADD_STOCK,
        payload,
        onSuccess,
        onError
    }
}

export const deleteStock = (payload, onSuccess, onError) => {
    return {
        type: types.DELETE_STOCK,
        payload,
        onSuccess,
        onError
    }
}

export const addStockToRecent = (payload) => {
    return {
        type: types.ADD_RECENTSTOCK,
        payload
    }
}

