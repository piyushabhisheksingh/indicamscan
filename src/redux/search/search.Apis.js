import { api } from '../config/api';
import { apiRoutes, BASE_API_URL } from '../config/routes';
import { base64 } from '../../utils/commonFunctions';
import { setLoading, removeLoading } from '../../components/common/LoadingIndicator'
import { services } from '../config/api'
/**
 * search, this will take one argument and it converts params to base64 and then it will call the search api
 */
export const search = (params) => {
    setLoading();
    /**
    * @param {string} params - params contains display segment and search query
    */
    let encoded = base64(params);
    return api(BASE_API_URL).get(apiRoutes.search + encoded)
        .then((response) => {
            removeLoading();
            if (response && response.data) {
                return { response };
            } else {
                return { response: null };
            }
        })
        .catch((error) => {
            removeLoading();
            return { response: error };
        });
};

export const addStock = (params) => {
    setLoading();
    services.apitype = 'addstock';
    return api(BASE_API_URL).post(apiRoutes.addStock(params.watchlistId, params.wtoken))
        .then((response) => {
            removeLoading();
            return response
        })
        .catch((error) => {
            removeLoading();
            return error
        })
}

export const addStockByWname = (params) => {
    setLoading();
    services.apitype = 'addStockByWname';
    return api(BASE_API_URL).post(apiRoutes.addStockByWname(params.watchlistName, params.wtoken))
        .then((response) => {
            removeLoading();
            return response
        })
        .catch((error) => {
            removeLoading();
            return error
        })
}

export const addRecentStock = (params) => {
    services.apitype = 'addstock';
    setLoading();
    return api(BASE_API_URL).post(apiRoutes.recentAddStock(params.watchlistName, params.wtoken))
        .then((response) => {
            removeLoading();
            return response
        })
        .catch((error) => {
            removeLoading();
            return error
        })
}

export const deleteStock = (params) => {
    services.apitype = 'deleteStock';
    setLoading();
    return api(BASE_API_URL).delete(apiRoutes.deleteStock(params.watchlistId, params.wtoken))
        .then((response) => {
            removeLoading();
            return response
        })
        .catch((error) => {
            removeLoading();
            return error
        })
}

/**
 * mock, this will take no arguments and it will call the news api
 */
/*export const mock = () => {
    const url = apiRoutes.mocks[getRandom()].news;
    return mockApi.get(url)
        .then((response) => {
            if (response && response.data) {
                return response;
            } else {
                return null;
            }
        })
        .catch((error) => {
            return error;
        });
}*/