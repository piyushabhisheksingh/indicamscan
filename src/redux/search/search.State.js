export default {
  search: {
    all: [],
    cash: [],
    currency: [],
    commodity: [],
    fno: [],
    selectedTab: 0,
    searchText: '',
    loading: { key: 0, state: false },
    news: [],
    index: 3,
    recentSearchVisible: true,
  }
};