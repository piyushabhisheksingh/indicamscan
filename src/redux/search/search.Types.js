/**
 * action type for search
 */
export const SEARCH = 'SEARCH';
/**
 * action type for selected tab
 */
export const SET_SELECTED_TAB = 'SET_SELECTED_TAB';
/**
 * action type for search text
 */
export const SET_SEARCH_TEXT = 'SET_SEARCH_TEXT'
/**
 * action type for success search api response
 */
export const SEARCH_SUCCESS = 'SEARCH_SUCCESS'
/**
 * action type for failure search api response
 */
export const SEARCH_FAILURE = 'SEARCH_FAILURE'
/**
 * action type for clearing search state
 */
export const CLEAR_SEARCH_STATE = 'CLEAR_SEARCH_STATE'
/**
 * action type for reatining data
 */
export const RETAIN_DATA = 'RETAIN_DATA'
/**
 * action type for clearing search result
 */
export const CLEAR_RESULT = 'CLEAR_RESULT'
/**
 * action type for success news api response
 */
export const NEWS_SUCCESS = 'NEWS_SUCCESS'

/**
 * action type for recent search 
 */
export const RECENT_SEARCH = "RECENT_SEARCH"


export const SET_ENABLE_LOG = 'SET_ENABLE_LOG'

export const ADD_STOCK = 'ADD_STOCK';

export const DELETE_STOCK = 'DELETE_STOCK';

/**
 * action type for addToStock api success response
 */
export const ADDSTOCK_SUCCESS = "ADDSTOCK_SUCCESS"
/**
 * action type for addToStock api failure response
 */
export const ADDSTOCK_FAILURE = "ADDSTOCK_FAILURE"

/**
* Action type for adding stock to recent list
*/
export const ADD_RECENTSTOCK = "ADD_RECENTSTOCK"

