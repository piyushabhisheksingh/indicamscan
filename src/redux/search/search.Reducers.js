import * as types from './search.Types';
import appState from '././search.State';
import lables from '../../utils/lables';
import { DeviceEventEmitter } from 'react-native';
import constants from '../../utils/appConstants'
import { eventName, LogEvents } from '../../../__analytics__/AppAnalytics';

/**
 * searchReducer, The reducer is a pure function that takes the previous state and an action, and returns the next state. (previousState, action) => nextState.
 */
const searchReducer = (state = appState.search, action) => {
    /**
    * @param {Object|string} state, action - state is the previousState and action is the particular action type.
    */
    switch (action.type) {
        case types.SEARCH: {

            if (action.prevIndex != action.mode) {
                if (action.prevIndex == 0) {
                    return {
                        ...state,
                        all: [],
                        news: [],
                        index: 3,
                        loading: { key: 0, state: true },
                        selectedTab: action.mode,
                        searchText: action.scrip,
                        recentSearchVisible: false

                    }
                }
                else if (action.prevIndex == 1) {
                    return {
                        ...state,
                        cash: [],
                        news: [],
                        index: 3,
                        loading: { key: 1, state: true },
                        selectedTab: action.mode,
                        searchText: action.scrip,
                        recentSearchVisible: false

                    }
                }
                else if (action.prevIndex == 2) {
                    return {
                        ...state,
                        fno: [],
                        news: [],
                        index: 3,
                        loading: { key: 2, state: true },
                        selectedTab: action.mode,
                        searchText: action.scrip,
                        recentSearchVisible: false

                    }
                }
                else if (action.prevIndex == 3) {
                    return {
                        ...state,
                        currency: [],
                        news: [],
                        index: 3,
                        loading: { key: 3, state: true },
                        selectedTab: action.mode,
                        searchText: action.scrip,
                        recentSearchVisible: false

                    }
                }
                else {
                    return {
                        ...state,
                        commodity: [],
                        news: [],
                        index: 3,
                        loading: { key: 4, state: true },
                        selectedTab: action.mode,
                        searchText: action.scrip,
                        recentSearchVisible: false

                    }
                }
            } else {
                return {
                    ...state,
                    all: [],
                    cash: [],
                    fno: [],
                    currency: [],
                    commodity: [],
                    news: [],
                    index: 3,
                    loading: { key: action.mode, state: true },
                    selectedTab: action.mode,
                    searchText: action.scrip,
                    recentSearchVisible: false

                }
            }
        }
        case types.SET_SELECTED_TAB: {

            if (state.searchText.length <= 2 && state.searchText.length > 0) {
                return {
                    ...state,
                    selectedTab: action.payload,
                    all: lables.minchar,
                    cash: lables.minchar,
                    currency: lables.minchar,
                    commodity: lables.minchar,
                    fno: lables.minchar,
                    recentSearchVisible: false

                }
            }
            else {
                return {
                    ...state,
                    selectedTab: action.payload,
                    all: [],
                    cash: [],
                    currency: [],
                    commodity: [],
                    fno: [],
                    recentSearchVisible: false

                }
            }
        }

        case types.RECENT_SEARCH: {
            // if (!state.recentSearchVisible){
            DeviceEventEmitter.emit(constants.RECENTSEARCH);
            // }
            return {
                ...state,
                recentSearchVisible: true
            }
        }

        case types.SET_SEARCH_TEXT:
            // if (state.recentSearchVisible){
            DeviceEventEmitter.emit(constants.SEARCH);
            // }
            return {
                ...state,
                searchText: action.payload,
                all: action.payload.length <= 1 ? lables.minchar : [],
                cash: action.payload.length <= 1 ? lables.minchar : [],
                currency: action.payload.length <= 1 ? lables.minchar : [],
                commodity: action.payload.length <= 1 ? lables.minchar : [],
                fno: action.payload.length <= 1 ? lables.minchar : [],
                loading: { key: action.key, state: false },
                recentSearchVisible: false
            }
        case types.SEARCH_SUCCESS: {
            let loading = state.loading;
            loading.state = false
            if (action.mode == 0) {
                return {
                    ...state,
                    all: action.data.result,
                    loading,
                    recentSearchVisible: false

                }
            }
            else if (action.mode == 1) {
                return {
                    ...state,
                    cash: action.data.result,
                    loading,
                    recentSearchVisible: false

                }
            }
            else if (action.mode == 2) {
                return {
                    ...state,
                    fno: action.data.result,
                    loading,
                    recentSearchVisible: false

                }
            }
            else if (action.mode == 3) {
                return {
                    ...state,
                    currency: action.data.result,
                    loading,
                    recentSearchVisible: false

                }
            }
            else {
                return {
                    ...state,
                    commodity: action.data.result,
                    loading,
                    recentSearchVisible: false

                }
            }
        }
        case types.SEARCH_FAILURE: {

            /**
             * Capturing Log Events for Search failure scenario
             */
            LogEvents(eventName.noSearchResult, {search_input: state.searchText})

            let loading = state.loading;
            loading.key = action.mode;
            loading.state = false
            if (action.mode == 0) {
                return {
                    ...state,
                    all: action.data,
                    loading,
                    recentSearchVisible: false

                }
            }
            else if (action.mode == 1) {
                return {
                    ...state,
                    cash: action.data,
                    loading,
                    recentSearchVisible: false

                }
            }
            else if (action.mode == 2) {
                return {
                    ...state,
                    fno: action.data,
                    loading,
                    recentSearchVisible: false

                }
            }
            else if (action.mode == 3) {
                return {
                    ...state,
                    currency: action.data,
                    loading,
                    recentSearchVisible: false

                }
            }
            else {
                return {
                    ...state,
                    commodity: action.data,
                    loading,
                    recentSearchVisible: false

                }
            }
        }
        case types.CLEAR_RESULT: {
            return {
                ...state,
                all: [],
                cash: [],
                currency: [],
                commodity: [],
                fno: [],
                news: [],
                index: 3,
                recentSearchVisible: true

            }
        }
        case types.CLEAR_SEARCH_STATE:
            return {

                all: [],
                cash: [],
                currency: [],
                commodity: [],
                fno: [],
                selectedTab: 0,
                searchText: '',
                loading: { key: 0, state: false },
                news: [],
                index: 3,
                recentSearchVisible: true


            }
        case types.NEWS_SUCCESS:
            return {
                ...state,
                news: action.data,
                index: action.index
            }
        case types.SET_ENABLE_LOG:
            return {
                ...state,
                enableLog: action.payload
            }
        default:
            return state
    }

}
export default searchReducer;