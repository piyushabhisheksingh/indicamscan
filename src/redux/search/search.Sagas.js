import { put, call, fork, takeLatest, delay, takeEvery } from 'redux-saga/effects';
import * as types from './search.Types';
import * as typesOne from '../watchlist/watchlist.Types';
import * as searchApis from './search.Apis';
import { setInfo, getInfo, getAppInfo } from '../../utils/localStorage'
import key from '../../utils/storagekey';
import store from '../store';
import * as watchlistApis from '../watchlist/watchlist.Apis';
import { convertToInt } from '../../utils/commonFunctions';
import { RECENT_SEARCH_RESULT } from '../recentSearch/recentSearch.Types';
import appConstants from '../../utils/appConstants';
import { perfEvents } from '../../../__analytics__/event'

/**
 * search, this function takes one argument. This function calls the search and api and then it returns the response
 */
export function* Search(action) {
    /**
    * @param {Object} action - actions contains request parameters
    */
    try {
        while (perfEvents.enableLog && perfEvents.eventDispatcher[1]) {
            perfEvents.eventDispatcher.shift();
        }
 
        yield delay(action.delay)
        if (perfEvents.enableLog && perfEvents.eventDispatcher[0]) {
            perfEvents.eventDispatcher[0].apiStartTime = Date.now();
            perfEvents.eventDispatcher[0].param =JSON.stringify(action.params);
            perfEvents.eventDispatcher[0].delay=200; 
        }
        const { params, mode } = action;
        if (action.scrip.length) {
            const result = yield call(searchApis.search, params);
            const data = result.response;
            if (data && data.data) {
                if (perfEvents.enableLog && perfEvents.eventDispatcher[0]) {
                    perfEvents.eventDispatcher[0].apiEndTime = Date.now();
                    perfEvents.eventDispatcher[0].doLog = true
                }
                yield put({ type: types.SEARCH_SUCCESS, data: data.data, mode})
                /*if (mode == 0 || mode == 1) {
                    for (let i = 0; i < data.data.result.length; i++) {
                        const item = data.data.result[i];
                        if (item.symbol == action.scrip) {
                            yield delay(500)
                            const newsData = yield call(searchApis.mock)
                            if (data.data.result && newsData && newsData.data && newsData.data.result && newsData.data.result.length > 0) {
                                let index = data.data.result.length > 3 ? 3 : data.data.result.length - 1
                                yield put({ type: types.NEWS_SUCCESS, data: newsData.data.result, index: index })
                            }
                            break;
                        }
                    }
                }*/

            } else {
                if (perfEvents.enableLog && perfEvents.eventDispatcher[0]) {
                    perfEvents.eventDispatcher.shift();
                }
                yield put({ type: types.SEARCH_FAILURE, data: data.result ? data.result : data.toString(), mode })
            }
        }
    } catch (error) {
        if (perfEvents.enableLog && perfEvents.eventDispatcher[0]) {
            perfEvents.eventDispatcher.shift();
        }
        const { mode } = action;
        yield put({ type: types.SEARCH_FAILURE, data: error.toString(), mode })
    }
}

/**
 * watchSearch, this is a watcher function for search action, it takes no arguments, it will call the search function and it always takes the latest call
 */
export function* watchSearch() {
    yield takeLatest(types.SEARCH, Search)
}

export function* AddStock(action) {
    try {
        
        // if (perfEvents.enableLog && perfEvents.eventDispatcher[0]) {
        //     perfEvents.eventDispatcher[0].apiStartTime = Date.now();
        //     perfEvents.eventDispatcher[0].param = action.payload;
        // }
        if (store.getState().watchlist.watchlistTab[0]) {
            const response = yield call(searchApis.addStock, action.payload);
            // if (perfEvents.enableLog && perfEvents.eventDispatcher[0]) {
            //     perfEvents.eventDispatcher[0].apiEndTime = Date.now();
            //     perfEvents.eventDispatcher[0].doLog = true
            // }
            if (response && response.data && response.data.Success) {
                let details = response.data.Success;
                let item = action.payload.item;
                let stcokresult = {
                    instrumentToken: Number(item.wtoken),
                    company: item.company,
                    expiryDate: details.expiryDate,
                    strikePrice: details.strikePrice,
                    symbol: details.symbol,
                    exchange: details.exchange,
                    optionType: details.optionType,
                    it: details.it,
                    tokenAddedLtp: details.tokenAddedLtp,
                    tokenAddedTime: details.tokenAddedTime,
                };
                yield put({ type: typesOne.ADDSTOCK_WATCHLIST_SUCCESS, payload: stcokresult, key: action.payload.watchlistId })
                action.onSuccess(stcokresult)
                const watchlistItems = store.getState().watchlist.watchlistItems;
                const watchlistTab = store.getState().watchlist.watchlistTab;
                yield call(setInfo, key.viewWatchlist + action.payload.watchlistId, watchlistItems)
                yield call(setInfo, key.listWatchlist, watchlistTab);
            } else {
                if (response && response.data && response.data.fault) {
                    action.onError(response.data.fault.message.toString())
                } else {
                    action.onError(response.toString())
                }
            }
        } else {
            // if (perfEvents.enableLog && perfEvents.eventDispatcher[0]) {
            //     perfEvents.eventDispatcher[0].apiStartTime = Date.now();
            //     perfEvents.eventDispatcher[0].param = action.payload;
            // }
            const response = yield call(searchApis.addStockByWname, action.payload)
            // if (perfEvents.enableLog && perfEvents.eventDispatcher[0]) {
            //     perfEvents.eventDispatcher[0].apiEndTime = Date.now();
            //     perfEvents.eventDispatcher[0].doLog = true
            // }

            if (response && response.data && response.data.Success) {
                let result = {
                    watchlistId: response.data.Success.watchlistId,
                    watchlistName: response.data.Success.watchlistName,
                    instrumentCount: 0
                };


                yield put({ type: typesOne.CREATE_WATCHLIST_SUCCESS, payload: result });
                let details = response.data.Success;
                let item = action.payload.item;
                let stcokresult = {
                    instrumentToken: Number(item.wtoken),
                    company: item.company,
                    expiryDate: details.expiryDate,
                    strikePrice: details.strikePrice,
                    symbol: details.symbol,
                    exchange: details.exchange,
                    optionType: details.optionType,
                    it: details.it,
                    tokenAddedLtp: details.tokenAddedLtp,
                    tokenAddedTime: details.tokenAddedTime,
                };

                yield put({ type: typesOne.ADDSTOCK_WATCHLIST_SUCCESS, payload: stcokresult, key: result.watchlistId })
                action.onSuccess(stcokresult, result)
                const watchlistItems = store.getState().watchlist.watchlistItems;
                const watchlistTab = store.getState().watchlist.watchlistTab;
                yield call(setInfo, key.viewWatchlist + action.payload.watchlistId, watchlistItems)
                yield call(setInfo, key.listWatchlist, watchlistTab);
            } else {
                if (response && response.data && response.data.fault) {
                    action.onError(response.data.fault.message.toString())
                } else {
                    action.onError(response.toString())
                }
            }
        }
    } catch (error) {
        action.onError(error.toString())
    }
}

export function* watchAddStock() {
    yield takeEvery(types.ADD_STOCK, AddStock)
}

export function* DeleteStock(action) {
    try {
        if (store.getState().watchlist.watchlistTab.length > 0) {
            // if (perfEvents.enableLog && perfEvents.eventDispatcher[0]) {
            //     perfEvents.eventDispatcher[0].apiStartTime = Date.now();
            //     perfEvents.eventDispatcher[0].param = action.payload;
            // }
            const response = yield call(searchApis.deleteStock, action.payload);
            // if (perfEvents.enableLog && perfEvents.eventDispatcher[0]) {
            //     perfEvents.eventDispatcher[0].apiEndTime = Date.now();
            //     perfEvents.eventDispatcher[0].doLog = true
            // }
            if (response && response.data && response.data.Success) {
                yield put({ type: typesOne.DELETESTOCK_WATCHLIST_SUCCESS, payload: action.payload, key: action.payload.watchlistId });
                action.onSuccess(action.payload.wtoken)
                const watchlistItems = store.getState().watchlist.watchlistItems;
                const watchlistTab = store.getState().watchlist.watchlistTab;
                yield call(setInfo, key.viewWatchlist + action.payload.watchlistId, watchlistItems)
                yield call(setInfo, key.listWatchlist, watchlistTab);
            } else {
                if (response && response.data && response.data.fault) {
                    action.onError(response.data.fault.message.toString())
                } else {
                    action.onError(response.toString())
                }

            }
        }
    } catch (error) {
        action.onError(error.toString())
    }
}

export function* watchDeleteStock() {
    yield takeEvery(types.DELETE_STOCK, DeleteStock)
}

export function* AddToRecent(action) {
    try {
        let token = []
        let cache = yield call(getAppInfo, key.recentWatchlist)

        if (cache) {
            token = cache.map((it) => {
                return convertToInt(it.instrumentToken)
            })
        }


        if (!token.includes(convertToInt(action.payload.item.wtoken))) {
            const response = yield call(searchApis.addRecentStock, action.payload);
            if (response && response.data && response.data.Success) {
                action.payload.item.instrumentToken = convertToInt(action.payload.item.wtoken)
                action.payload.item.expiryDate = action.payload.item.expiry
                action.payload.item.strikePrice = action.payload.item.strikeprice
                action.payload.item.optionType = action.payload.item.options
                let list = []
                if (cache) {
                    list = [action.payload.item, ...cache]
                    yield call(setInfo, key.recentWatchlist, list)
                } else {
                    list = [action.payload.item]
                    yield call(setInfo, key.recentWatchlist, list)
                }

                yield put({ type: RECENT_SEARCH_RESULT, payload: list })
            }
        }

    } catch (error) { 
    }
}
export function* watchRecentStocks() {
    yield takeEvery(types.ADD_RECENTSTOCK, AddToRecent)
}

