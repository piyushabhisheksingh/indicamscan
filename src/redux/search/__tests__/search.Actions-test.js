
import * as search from '../search.Actions'

const onSuccess = jest.fn()
const onError = jest.fn()

describe('Search Actions', () => {
    it('1. verify search action', () => {
        const result = search.search(1, "ab", 0)
        expect(result).toMatchSnapshot();
    });
    it('2. verify clearSearchState action', () => {
        const result = search.clearSearchState()
        expect(result).toMatchSnapshot();
    });
    it('3. verify clearState action', () => {
        const result = search.clearState()
        expect(result).toMatchSnapshot();
    });
    it('4. verify setSearchText action', () => {
        const result = search.setSearchText("ab")
        expect(result).toMatchSnapshot();
    });
    it('5. verify setTab action', () => {
        const result = search.setTab(1)
        expect(result).toMatchSnapshot();
    });
    it('6. verify search action', () => {
        const result = search.search(0, "ab", 1)
        expect(result).toMatchSnapshot();
    });
    it('7. verify search action', () => {
        const result = search.search(2, "ab", 3)
        expect(result).toMatchSnapshot();
    });
    it('8. verify search action', () => {
        const result = search.search(3, "ab", 4)
        expect(result).toMatchSnapshot();
    });
    it('9. verify search action', () => {
        const result = search.search(4, "ab", 2)
        expect(result).toMatchSnapshot();
    });

    it('10. verify addStock action', () => {
        let payload = {watchlistId : 3000403, watchlistName: 'WatchlistOne', wtoken: 727}
        const result = search.addStock(payload, onSuccess, onError)
        expect(result).toMatchSnapshot();
    })
    it('11. verify addStockToRecent action', () => {
        let payload = {watchlistId : 3000403, watchlistName: 'recent', wtoken: 727}
        const result = search.addStockToRecent(payload)
        expect(result).toMatchSnapshot();
    })
    it('12. verify deleteStock action', () => {
        let payload = {watchlistId : 3000403, watchlistName: 'WatchlistOne', wtoken: 727}
        const result = search.deleteStock(payload, onSuccess, onError)
        expect(result).toMatchSnapshot();
    })
});