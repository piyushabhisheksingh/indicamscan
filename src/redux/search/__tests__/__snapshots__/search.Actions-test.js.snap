// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Actions 1. verify search action 1`] = `
Object {
  "delay": undefined,
  "event": null,
  "mode": 1,
  "params": "AB,_EQ",
  "prevIndex": 0,
  "scrip": "ab",
  "type": "SEARCH",
}
`;

exports[`Search Actions 2. verify clearSearchState action 1`] = `
Object {
  "type": "CLEAR_SEARCH_STATE",
}
`;

exports[`Search Actions 3. verify clearState action 1`] = `
Object {
  "type": "CLEAR_RESULT",
}
`;

exports[`Search Actions 4. verify setSearchText action 1`] = `
Object {
  "key": undefined,
  "payload": "ab",
  "type": "SET_SEARCH_TEXT",
}
`;

exports[`Search Actions 5. verify setTab action 1`] = `
Object {
  "payload": 1,
  "type": "SET_SELECTED_TAB",
}
`;

exports[`Search Actions 6. verify search action 1`] = `
Object {
  "delay": undefined,
  "event": null,
  "mode": 0,
  "params": "AB",
  "prevIndex": 1,
  "scrip": "ab",
  "type": "SEARCH",
}
`;

exports[`Search Actions 7. verify search action 1`] = `
Object {
  "delay": undefined,
  "event": null,
  "mode": 2,
  "params": "AB,_FNO",
  "prevIndex": 3,
  "scrip": "ab",
  "type": "SEARCH",
}
`;

exports[`Search Actions 8. verify search action 1`] = `
Object {
  "delay": undefined,
  "event": null,
  "mode": 3,
  "params": "AB,CUR_",
  "prevIndex": 4,
  "scrip": "ab",
  "type": "SEARCH",
}
`;

exports[`Search Actions 9. verify search action 1`] = `
Object {
  "delay": undefined,
  "event": null,
  "mode": 4,
  "params": "AB,COM_",
  "prevIndex": 2,
  "scrip": "ab",
  "type": "SEARCH",
}
`;

exports[`Search Actions 10. verify addStock action 1`] = `
Object {
  "onError": [MockFunction],
  "onSuccess": [MockFunction],
  "payload": Object {
    "watchlistId": 3000403,
    "watchlistName": "WatchlistOne",
    "wtoken": 727,
  },
  "type": "ADD_STOCK",
}
`;

exports[`Search Actions 11. verify addStockToRecent action 1`] = `
Object {
  "payload": Object {
    "watchlistId": 3000403,
    "watchlistName": "recent",
    "wtoken": 727,
  },
  "type": "ADD_RECENTSTOCK",
}
`;

exports[`Search Actions 12. verify deleteStock action 1`] = `
Object {
  "onError": [MockFunction],
  "onSuccess": [MockFunction],
  "payload": Object {
    "watchlistId": 3000403,
    "watchlistName": "WatchlistOne",
    "wtoken": 727,
  },
  "type": "DELETE_STOCK",
}
`;
